app/Models/Rating.php

```diff
@@ -48,4 +62,92 @@ public function author()
     {
         return $this->belongsTo(\App\Models\User::class);
     }
+
+    /**
+     * Determines if the rating instance has a category given by the passed parameter
+     * @param $bitMask
+     * bitmask param should always be one of the bitmask constants defined on this model
+     * @return bool
+     */
+    public function hasCategory($bitMask)
+    {
+        return ($this->categories & $bitMask) === $bitMask;
+    }
+
+    /**
+     * Adds a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function addCategory($bitMask)
+    {
+        $this->categories |= $bitMask;
+    }
+
+    /**
+     * Removes a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function removeCategory($bitMask)
+    {
+        $this->categories &= ~$bitMask;
+    }
+
+    /**
+     * Returns category labels. Not recommended for large batches of requests, instead use hasCategory with bitmask
+     * @returns Collection of strings corresponding to the rating categories of this entity
+     */
+    public function getCategoryLabels()
+    {
+        $labels = collect();
+        $bitmasks = collect([
+            self::CREATIVE_BITMASK,
+            self::FLEXIBLE_BITMASK,
+            self::FRIENDLY_BITMASK,
+            self::HELPFUL_BITMASK,
+            self::PREPARED_BITMASK,
+        ]);
+
+        foreach ($bitmasks as $mask) {
+            if ($this->hasCategory($mask)) {
+                $labels->push($this->getCategoryLabelFromMask($mask));
+            }
+        }
+
+        return $labels;
+    }
+
+    /**
+     * Get category label corresponding to a bitmask
+     * @param $mask integer The bitmask
+     * @return string The Label
+     */
+    private function getCategoryLabelFromMask($mask)
+    {
+        $label = "";
+        switch ($mask) {
+            case self::CREATIVE_BITMASK:
+                $label = self::CREATIVE_LABEL;
+                break;
+
+            case self::FLEXIBLE_BITMASK:
+                $label = self::FLEXIBLE_LABEL;
+                break;
+
+            case self::FRIENDLY_BITMASK:
+                $label = self::FRIENDLY_LABEL;
+                break;
+
+            case self::HELPFUL_BITMASK:
+                $label = self::HELPFUL_LABEL;
+                break;
+
+            case self::PREPARED_BITMASK:
+                $label = self::PREPARED_LABEL;
+                break;
+
+            default:
+                $label = "";
+        }
+        return $label;
+    }
```

Comment:
Lets also add some query scopes since we'll need to be able to filter on this in #208, The article you referenced [lays out how to do that](https://aaronfrancis.com/2021/bitmasking-in-laravel-and-mysql#bitmasks-in-mysql) pretty nicely

If you need more info in general about query scopes, [the docs also have some good info](https://laravel.com/docs/8.x/eloquent#query-scopes)

Reply:
all done!